### PR TITLE
Cloud init disable

### PR DESCRIPTION
**Pull request by diranetafen**
Disable Cloud-init to prevent the regeneration of network interface config

### CODE REVIEW

File: kayobe/tests/unit/cli/test_commands.py [hunk 1]
```diff
@@ -466,6 +466,7 @@ def test_overcloud_host_configure(self, mock_kolla_run, mock_run,
                     "ansible/network.yml",
                     "ansible/sysctl.yml",
                     "ansible/disable-glean.yml",
+                    "ansible/disable-cloud-init.yml"
```
Comment (markgoddard) [inline kayobe/tests/unit/cli/test_commands.py]: Missing the trailing semicolon - this is why the test is failing.

File: ansible/roles/disable-cloud-init/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  with_items: "{{ cloud_init_services.stdout_lines }}"
+  become: True
+
+
+- name: Ensure Cloud-init artifacts are removed
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - "/usr/lib/systemd/system/cloud-init@.service"
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/tasks/main.yml]: I don't see this file on my system.

File: ansible/roles/disable-cloud-init/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  register: cloud_init_services
+  changed_when: False
+
+- name: Ensure Cloud-init services are stopped and disabled
+  service:
+    name: "{{ item.split()[0] }}"
+    state: stopped
+    enabled: no
+  with_items: "{{ cloud_init_services.stdout_lines }}"
+  become: True
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/tasks/main.yml]: Also notify the systemd daemon here?

File: ansible/roles/disable-cloud-init/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Check for existing CLoud-init systemd services
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/tasks/main.yml]: nit: CLoud-init

File: ansible/roles/disable-cloud-init/handlers/main.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Reload systemd daemon
+  command: systemctl daemon-reload
+  become: True
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/handlers/main.yml]: nit: you can use http://docs.ansible.com/ansible/latest/systemd_module.html.

**Review by markgoddard — CHANGES_REQUESTED** [review]
Thanks for this Dirane.

I recently spent some time ensuring that cloud-init works well both for the seed and the overcloud hosts. This change would prevent cloud-init from working. At the least we need to put this behind a flag, e.g. `disable_cloud_init: False`. See https://github.com/stackhpc/kayobe/issues/111 and https://github.com/stackhpc/kayobe/issues/112.

Another (smaller) concern I have is that by disabling cloud-init after the first boot, we require that first boot to be successful. I had a few problems with getting glean and cloud-init to work together. Wouldn't a better solution be to disable or uninstall cloud-init in the image so that it never runs?

Finally, have you tried other methods of disabling cloud-init's network config, such as those listed here: http://cloudinit.readthedocs.io/en/latest/topics/network-config.html#disabling-network-configuration? The simplest I think would be to write a file to /etc/cloud/cloud.cfg.d/disable-net.cfg, containing this:

```
network:
  config: disabled
```

File: ansible/roles/disable-cloud-init/handlers/main.yml (hunk 5 above)
Comment (diranetafen) [reply ansible/roles/disable-cloud-init/handlers/main.yml]: Thanks for the link ! i will use systemd: daemon_reload=yes

File: ansible/roles/disable-cloud-init/tasks/main.yml (hunk 4 above)
Comment (diranetafen) [reply ansible/roles/disable-cloud-init/tasks/main.yml]: :-) changed

File: ansible/roles/disable-cloud-init/tasks/main.yml (hunk 2 above)
Comment (diranetafen) [reply ansible/roles/disable-cloud-init/tasks/main.yml]: I have those files in my environment

File: kayobe/tests/unit/cli/test_commands.py (hunk 1 above)
Comment (diranetafen) [reply kayobe/tests/unit/cli/test_commands.py]: corrected

File: ansible/roles/disable-cloud-init/handlers/main.yml [hunk 6]
```diff
@@ -1,4 +1,3 @@
 ---
 - name: Reload systemd daemon
-  command: systemctl daemon-reload
-  become: True
+  systemd: daemon_reload=yes
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/handlers/main.yml]: Still require become.

Comment (markgoddard) [reply ansible/roles/disable-cloud-init/handlers/main.yml]: nit: prefer YAML syntax:

```
systemd:
  daemon_reload: yes
```

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Comment by diranetafen** [comment]
Hi @markgoddard 
i'm still working on my pull request https://github.com/stackhpc/kayobe/pull/122
And I have a difficulty to understand exactly the problem with it! Because my playbook runs when the overcloud hosts are configured, so the first boot will complete successfully (after provision overcloud).

**Comment by diranetafen** [comment]
I will make some changes in my code and test it based on this article : https://fatmin.com/2017/10/19/how-to-disable-cloud-init-in-a-rhel-cloud-image/

File: ansible/roles/disable-cloud-init/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+- name: Disable cloud init service
+  file:
+    path: /etc/cloud/cloud-init.disabled 
+    state: touch
+    mode: "u=rw,g=r,o=r" 
+  when: disable_cloud_init | bool
```
Comment (markgoddard) [inline ansible/roles/disable-cloud-init/tasks/main.yml]: nit: Could you move this `when` to the disable-cloud-init.yml playbook, like this:

```
role: disable-cloud-init
when: disable_cloud_init | bool
```

File: ansible/group_vars/all/overcloud [hunk 8]
```diff
@@ -16,6 +16,9 @@ overcloud_groups: >
 # should not be added to the inventory.
 overcloud_group_hosts_map: {}
 
+# To prevent some network issues you can choose to disable cloud-init
+disable_cloud_init: False
```
Comment (markgoddard) [inline ansible/group_vars/all/overcloud:20]: Please update etc/kayobe/overcloud.yml.

**Review by markgoddard — CHANGES_REQUESTED** [review]
This is looking better, thanks.

File: ansible/group_vars/all/overcloud (hunk 8 above)
Comment (diranetafen) [reply ansible/group_vars/all/overcloud:20]: Done !

**Review by markgoddard — APPROVED** [review]
I made a small change to etc/kayobe/overcloud.yml to comment out the variable and remove its value. Otherwise, looks good - thanks!

**Comment by diranetafen** [comment]
Ok @markgoddard  thanks

**Pull request merged**